### DISCUSSION TITLE
"Enable SMTP SSL" on Server Settings

### DISCUSSION THREAD

**Discussion by james7342** [dnnsoftware/Dnn.Platform#5753] [Q&A] [answered]
I think I may have stumbled across an issue though I thought I would start it as a discussion first. I currently use PowerDNN/Managed.com SMTP Relay service for sending messages. Installed HotCakesModule and noticed an error when it was sending emails. Worked with Will Strohl on it and I think may have found an issue. 

https://github.com/HotcakesCommerce/hotcakes-commerce-core/issues/453

I'm not convinced that the "Enable SMTP SSL" is actually working properly or at least enforcing SSL. Managed.com confirmed their SMTP relay doesn't support SSL, but with this option checked I can still send mail without issue. Does DNN try SSL first then fallback to non SSL if it fails? The only module that is having an issue is HCC. All of the other DNN emails work regardless if this option is toggled on/off.

**Comment by bdukes** [comment, marked as answer]
The MailKit provider does not do anything with the _Enable SMTP SSL_ configuration option. Instead, it uses an "auto" setting within MailKit, which detects the correct mode based on the port. I believe we needed to do this to support STARTTLS automatically (at least, without then also exposing other configuration that the older mail provider couldn't support).
https://github.com/dnnsoftware/Dnn.Platform/blob/90e6ac46ea59a933c65df96f842ffbb974b45b0b/DNN%20Platform/Library/Services/Mail/MailKitMailProvider.cs#L43

**Reply by bdukes** [reply]
In contrast, the older "core" mail provider does set [the `SmtpClient.EnableSsl` property](https://learn.microsoft.com/en-us/dotnet/api/system.net.mail.smtpclient.enablessl?view=netframework-4.8) based on that configuration.
https://github.com/dnnsoftware/Dnn.Platform/blob/90e6ac46ea59a933c65df96f842ffbb974b45b0b/DNN%20Platform/Library/Services/Mail/CoreMailProvider.cs#L275

**Reply by james7342** [reply]
Ok. I'm using MailKit therefore the Auto setting. Still something I think may not be working correctly with it. Or my understanding is incorrect. We all know default is port 25 and 587 (SSL). What I would expect is if MailKit is using the port to determine the connection method. If no port is provided in the SMTP Mail Server settings, Hostname.domain.com, then it would default to non secure port 25. Though if the same configuration is applied with a port Hostname.domain.com:587 then it should use SSL. Is that the correct understanding the of "Auto" setting?

I tried doing both with my relay services and both work. If "auto" is working correctly I would have suspected mail sent using port 587 would fail as SSL is not enabled on the relay.

If I'm not mistaken, HCC using the SMTP settings from DNN, but is using .net provider to send the messages. The same one that is used with the DNN CoreMail.

Maybe an enhancement or note update to indicate "Enable SMTP SSL" has no impact if using MailKit. I obviously didn't know that and thought it was using SSL with it toggled on.

**Reply by bdukes** [reply]
According to https://stackoverflow.com/a/57008444/2688:
>MailKit treats port `587` the same as it treats `25`. In other words, it treats port `25` and `587` as a plain-text connection port but will switch to SSL/TLS if requested to do so via `STARTTLS`.

I agree that a note would be helpful.

**Reply by james7342** [reply]
That explains what I'm seeing and why my configuration/testing is successful.

Better yet on the enhancement- Check if MailKit is being used and disable the "Enable SMTP SSL" option and cover the reason in the information popup. :-)

**Reply by james7342** [reply]
@bdukes do you want me to open a issue for an enhancement and link this discussion?

**Reply by bdukes** [reply]
I just submitted https://github.com/dnnsoftware/Dnn.Platform/pull/5756 to add a note

**Reply by james7342** [reply]
Thank you..
